test/helpers/SessionAnswersHelperSpec.scala

```diff
@@ -1146,4 +1156,53 @@ class SessionAnswersHelperSpec extends SpecBase {
       }
     }
   }
+
+  "getPreviousUploadsFileNames" should{
+    "return Future[String] " in {
+      val mockRepository = mock(classOf[UploadJourneyRepository])
+      implicit val ec: ExecutionContext = injector.instanceOf[ExecutionContext]
+      val fakeRequestForOtherJourney: UserRequest[AnyContent] = fakeRequestConverter(fakeRequestWithCorrectKeys.withSession(
+        SessionKeys.reasonableExcuse -> "other",
+        SessionKeys.hasConfirmedDeclaration -> "true",
+        SessionKeys.whyReturnSubmittedLate -> "This is a reason.",
+        SessionKeys.whenDidBecomeUnable -> "2022-01-02",
+        SessionKeys.journeyId -> "4321"
+      ))
+      val callBackModel: UploadJourney = UploadJourney(
+        reference = "ref1",
+        fileStatus = UploadStatusEnum.READY,
+        downloadUrl = Some("download.file/url"),
+        uploadDetails = Some(UploadDetails(
+          fileName = "file1.txt",
+          fileMimeType = "text/plain",
+          uploadTimestamp = LocalDateTime.of(2018, 1, 1, 1, 1),
+          checksum = "check1234",
+          size = 2
+        ))
+      )
+      when(mockRepository.getUploadsForJourney(Some("4321"))).thenReturn(Future(Option(Seq(callBackModel))))
+      val  helper = new SessionAnswersHelper(mockRepository)
+      whenReady(helper.getPreviousUploadsFileNames()(fakeRequestForOtherJourney)){
+       result => result shouldBe "file1.txt"
+     }
+    }
+  }
+
+  "getContentWithExistingUploadFileNames" should{
+    "return Future[Seq[(String, String, String)]] " in {
+      implicit val ec: ExecutionContext = injector.instanceOf[ExecutionContext]
+      val fakeRequestForOtherJourney: UserRequest[AnyContent] = fakeRequestConverter(fakeRequestWithCorrectKeys.withSession(
+        SessionKeys.reasonableExcuse -> "other",
+        SessionKeys.hasConfirmedDeclaration -> "true",
+        SessionKeys.whyReturnSubmittedLate -> "This is a reason.",
+        SessionKeys.whenDidBecomeUnable -> "2022-01-02",
+        SessionKeys.journeyId -> "4321"
+      ))
+      val mockHelper = mock(classOf[SessionAnswersHelper])
```

Comment:
We shouldn't actually mock the instance of the class we're testing, there is an already injected instance of SessionAnswersHelper at the top of this file

Reply:
Well the Mockito.verify method expects a mocked value else the test fails with  org.mockito.exceptions.misusing.NotAMockException

Reply:
Ok understandable it does that. But ideally, we want SessionAnswersHelper to be a real instance and not a mocked one, because with a mocked class you can essentially tell it to do anything and not test _its own_ functionality

Reply:
Implemented the test case with the appropriate strategy